main.nf

```diff
@@ -390,9 +391,18 @@ workflow {
 
     // STEP 1: MAPPING READS TO REFERENCE GENOME WITH BWA MEM
 
-    BWAMEM2_MEM(reads_input, bwa, fasta, fai, params.modules['bwamem2_mem'])
+    bam_bwamem2          = Channel.empty()
+    bwa_mem_out          = Channel.empty()
 
-    bam_bwamem2 = BWAMEM2_MEM.out
+    if (params.aligner == "bwa-mem") {
+        BWA_MEM(reads_input, bwa, fasta, fai, params.modules['bwa_mem'])
+        bwa_mem_out = BWA_MEM.out.bam
+    } else {
+        BWAMEM2_MEM(reads_input, bwa, fasta, fai, params.modules['bwamem2_mem'])
+        bam_bwamem2 = BWAMEM2_MEM.out
+    }
+
+    bam_bwamem2 = bam_bwamem2.mix(bwa_mem_out)
```

Comment:
I think we should use just bam_bwa or something. If `bwamem2` was not used for mapping, the channel will still be called that for the remainder of the pipeline

Reply:
Or will this not work because of empty channels or what not?I suppose we can also work on naming as we go, since all the things are changing so much still